test/index.test.ts

```diff
@@ -129,24 +129,8 @@ test('handles text body with content-type init options', async (test) => {
   test.deepEqual(axiosBody.headers['content-type'], expectedBody.headers['content-type']);
 });
 
-test('handles json body init options', async (test) => {
```

Comment:
Can this test be added back, or did it stop working?

Reply:
Dropping support for JSON in body is scary to me. Since the tests run both node-fetch and also axios-fetch based requests. Does that mean that node-fetch already supported the non-standard "JSON in body" approach? If so, I'd prefer to avoid breaking that feature, I'm almost positive that some code already depends on it working.

Reply:
Sooo 😅 I wanted to implement it based on your request, but... 

I have found out that actually, that use-case does not work even with the current master version 😉 The reason why it seems it works is that the test is written wrongly. I have printed out the body that `nock` receives and as per my suspicion I got:

```
Body:  [object Object]
Body:  [object Object]
```

And that is most probably because you use the `String(init.body)` and the tests is passing because you don't actually assert that the received body is the same as the sent one, but that the received bodies from both `fetch` and `axios-fetch` are the same which they are... `[object Object]`.

So I think it is safe to "break" this feature because nobody could have used it 😉

Reply:
Oh wow. That's a great insight! Sorry I missed this last comment. I just did a pass through my GitHub notifications and saw it. Let me check how close this is to mergable.